File: plugin/octo_beat_plugin/ios/Classes/core/utils/Role.swift
//
//  Role.swift
//  OctoBeat-clinic-ios
//
//  Created by Manh Tran on 2021-04-03.
//  Copyright © 2021 itr.vn. All rights reserved.
//

import Foundation

enum Role: String, CaseIterable {
    case CLINIC_TECHNICIAN = "Clinic Technician"
    case CLINIC_PHYSICIAN = "Clinic Physician"
    case PATIENT = "Patient";
    
    public static func get(role: String) -> Role {
        for r in  Role.allCases {
            if(role == r.rawValue) {
                return r
            }
        }
        return Role.PATIENT
    }
}

File: plugin/octo_beat_plugin/android/src/main/kotlin/com/octo/octo_beat_plugin/repository/enum_/Role.kt
package com.octo.octo_beat_plugin.repository.enum_

enum class Role(var value: String) {
    CLINICIAN_TECHNICIAN("Clinic Technician"),
    CLINICIAN_PHYSICIAN("Clinic Physician"),
    PATIENT("patient");

    companion object {
        fun get(role: String?): Role {
            for (r in values()) {
                if (r.value == role) {
                    return r
                }
            }
            return PATIENT
        }
    }
}